### PR TITLE
[SW-229036] out of box fp8 Dynamic scaling support

### PR DESCRIPTION
**Pull request by xuechendi**
https://jira.habana-labs.com/browse/SW-229036

Add a new env var: VLLM_HPU_FORCE_CHANNEL_FP8, default is 1 => all fp8 block scale will be converted to channel scale.
For running with INC static quant, need to set VLLM_HPU_FORCE_CHANNEL_FP8=0

This PR has depenencies on https://github.com/HabanaAI/vllm-hpu-extension/pull/185

1. support fp8 models with "compressed-tensor" quant_method
    https://huggingface.co/meta-llama/Llama-4-Maverick-17B-128E-Instruct-FP8
    https://huggingface.co/RedHatAI/Llama-4-Scout-17B-16E-Instruct-FP8-dynamic
2. support fp8 models with "block_scale fp8" quant_method
    https://huggingface.co/Qwen/Qwen3-30B-A3B-FP8
    https://huggingface.co/Qwen/Qwen3-235B-A22B-FP8
    https://huggingface.co/deepseek-ai/DeepSeek-R1
3. Add 4 new UT to test both compressed-tensor and block_scale fp8
    UT1: test compressed_tensor with llama4-scout
    UT2: test block scale fp8 with qwen3-30B - block_scale convert to per-channel during model loading
    UT3: test block scale fp8 with qwen3-30B - runtime dequent fp8 to bf16
    UT4: test block scale fp8 with qwen3-30B - INC convert block fp8 to per channel [Same way we did for deepseek inc path]

Quality check:
1. verified accuracy
** Deepseek **
vllm (pretrained=/mnt/weka/data/pytorch/DeepSeek-R1/,tensor_parallel_size=8,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=4096,use_v2_block_manager=True,dtype=bfloat16,enable_expert_parallel=True,max_num_seqs=128), gen_kwargs: (None), limit: 256.0, num_fewshot: 5, batch_size: 128

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.9688|±  |0.0109|
|     |       |strict-match    |     5|exact_match|↑  |0.9688|±  |0.0109|

** Qwen3-30B **
vllm (pretrained=/mnt/weka/llm/Qwen3-30B-A3B-FP8/,tensor_parallel_size=8,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=4096,use_v2_block_manager=True,dtype=bfloat16,enable_expert_parallel=True,max_num_seqs=128), gen_kwargs: (None), limit: 256.0, num_fewshot: 5, batch_size: 128

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8828|±  |0.0201|
|     |       |strict-match    |     5|exact_match|↑  |0.9297|±  |0.0160|

** llama4-Maverick **
vllm (pretrained=/mnt/weka/llm/Llama-4-Maverick-17B-128E-Instruct-FP8/,tensor_parallel_size=8,max_model_len=4096,max_num_seqs=1024,gpu_memory_utilization=0.8,use_v2_block_manager=True,dtype=bfloat16,max_gen_toks=2048,disable_log_stats=False,enable_expert_parallel=True), gen_kwargs: (None), limit: 0.1, num_fewshot: 0, batch_size: 128

|       Tasks       |Version|    Filter    |n-shot|  Metric   |   |Value |   |Stderr|
|-------------------|------:|--------------|-----:|-----------|---|-----:|---|-----:|
|mmlu_pro           |      2|custom-extract|      |exact_match|↑  |0.8089|±  |0.0109|
| - biology         |      1|custom-extract|     0|exact_match|↑  |0.9167|±  |0.0328|
| - business        |      1|custom-extract|     0|exact_match|↑  |0.8354|±  |0.0420|
| - chemistry       |      1|custom-extract|     0|exact_match|↑  |0.8772|±  |0.0309|
| - computer_science|      1|custom-extract|     0|exact_match|↑  |0.8293|±  |0.0595|
| - economics       |      1|custom-extract|     0|exact_match|↑  |0.8941|±  |0.0336|
| - engineering     |      1|custom-extract|     0|exact_match|↑  |0.7423|±  |0.0446|
| - health          |      1|custom-extract|     0|exact_match|↑  |0.7561|±  |0.0477|
| - history         |      1|custom-extract|     0|exact_match|↑  |0.5897|±  |0.0798|
| - law             |      1|custom-extract|     0|exact_match|↑  |0.6126|±  |0.0464|
| - math            |      1|custom-extract|     0|exact_match|↑  |0.9559|±  |0.0177|
| - other           |      1|custom-extract|     0|exact_match|↑  |0.6882|±  |0.0483|
| - philosophy      |      1|custom-extract|     0|exact_match|↑  |0.7200|±  |0.0641|
| - physics         |      1|custom-extract|     0|exact_match|↑  |0.9077|±  |0.0255|
| - psychology      |      1|custom-extract|     0|exact_match|↑  |0.7875|±  |0.0460|

| Groups |Version|    Filter    |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|--------------|------|-----------|---|-----:|---|-----:|
|mmlu_pro|      2|custom-extract|      |exact_match|↑  |0.8089|±  |0.0109|

### CODE REVIEW

**Comment by xuechendi** [comment]
@yiliu30 @jikunshang please help to review

**Comment by xuechendi** [comment]
@kzawora-intel  @madamczyk-intel @michalkuligowski

**Review by yiliu30 — APPROVED** [review]
LGTM

**Review by jikunshang — APPROVED** [review]
LGTM.
